src/Chapter2.hs

```diff
@@ -349,7 +364,9 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf l = take halfLenght l
+  where
+    halfLenght = div (length l) 2
```

Comment:
Great solution! Could you also add a type signature to this function? 🙂

Reply:
Yes, sure. I forgot about it 🤦‍♂️